File: 0739 daily-temperatures/solution.java
class Solution {
    public int[] dailyTemperatures(int[] t) {
        int[] st = new int[t.length];
        int res[]=new int[t.length];
        int top=-1;
        for(int i=0;i<t.length;i++) {
            while(top!=-1 && t[st[top]]<t[i]){
               res[st[top]]=i-st[top--];
            }
            st[++top]=i;
        }
        return res;
    }
}

File: 0739 daily-temperatures/solution.rs
impl Solution {
    pub fn daily_temperatures(t: Vec<i32>) -> Vec<i32> {
        let n = t.len();
        let mut st: Vec<usize> = Vec::new();
        let mut ans = vec![0; n];
        for i in 0..n {
            while !st.is_empty() && t[*st.last().unwrap()] < t[i] {
                if let Some(top) = st.pop() {
                     ans[top] = (i-top) as i32;
                }
            }
            st.push(i);
        }
        ans
    }
}
